projects/margin-zero/index.js

```diff
@@ -0,0 +1,109 @@
+const { cachedGraphQuery } = require("../helper/cache");
+
+const chainConfig = {
+  146: {
+    chainName: "sonic",
+    subgraph: "https://api.goldsky.com/api/public/project_cm58q8wq01kbk01ts09lc52kp/subgraphs/mz-subgraph/main/gn"
+  },
+}
+
+function LiquidityRangesQuery(first, skip) {
+  return `{ liquidityRanges(first: ${first}, skip: ${skip}) {pool liquidity tickLower tickUpper }}`
+}
+
+const slot0Abi =
+  "function slot0() view returns (uint160 sqrtPriceX96, int24 tick, uint16 observationIndex, uint16 observationCardinality, uint16 observationCardinalityNext, uint8 feeProtocol, bool unlocked)";
+
+const token0Abi =
+  "function token0() view returns (address)";
+
+const token1Abi =
+  "function token1() view returns (address)";
+
+
+function unwrapLiquidity(tickLower, tickUpper, currTick, liquidity) {
+  const tickToPrice = (tick) => 1.0001 ** tick;
+  const bottomTick = +tickLower;
+  const topTick = +tickUpper;
+  const tick = +currTick;
+  const sa = tickToPrice(bottomTick / 2);
+  const sb = tickToPrice(topTick / 2);
+
+  let amount0 = 0;
+  let amount1 = 0;
+
+  if (tick < bottomTick) {
+    amount0 = (liquidity * (sb - sa)) / (sa * sb);
+  } else if (tick < topTick) {
+    const price = tickToPrice(tick);
+    const sp = price ** 0.5;
+
+    amount0 = (liquidity * (sb - sp)) / (sp * sb);
+    amount1 = liquidity * (sp - sa);
+  } else {
+    amount1 = liquidity * (sb - sa);
+  }
+
+  return { amount0, amount1 };
+}
+
+
+async function getPoolTokens(api, pool, chainName) {
+  const [token0, token1] = await Promise.all([
+    api.call({ abi: token0Abi, target: pool }),
+    api.call({ abi: token1Abi, target: pool })
+  ])
+
+  return {
+    token0,
+    token1
+  }
+}
+
+
+async function tvl(sdk) {
+  if (sdk.api.chainId && chainConfig[sdk.api.chainId]) {
+    const config = chainConfig[sdk.api.chainId]
+    const subgraphUrl = config.subgraph
+    const chainName = config.chainName
+
+    let cummulativeLiquidityRanges = []
+
+    let counter = 0;
+    while (true) {
+      const { liquidityRanges } = await cachedGraphQuery('marign-zero/tvl', subgraphUrl, LiquidityRangesQuery(1000, 1000 * counter))
```

Comment:
you should use `fetchById` here (also, update the query), like https://github.com/0xradishmz/DefiLlama-Adapters/blob/pr/0xradishmz/12971/projects/panoptic/index.js#L63

noticed that panoptic & this adapter have the same bug, if we use cached graph data, we will end up with incorrect liquidity data & wrong tvl, but using tokenIds fixing it (so we fetch liquidity on chain)